diff --git a/install.sh b/install.sh
index fbb83cc..38acb15 100755
--- a/install.sh
+++ b/install.sh
@@ -38,28 +38,18 @@ MOD_SRC_DIR="/usr/src/$MODULE_NAME-$VERSION"
 
 if ! (dkms status 2>/dev/null | grep -q "$MODULE_NAME/${VERSION}.*installed"); then # if the module is already installed in DKMS
 
-  # For Debian
-  if command -v update-secureboot-policy >/dev/null 2>&1; then
-    update-secureboot-policy --new-key
-    update-secureboot-policy --enroll-key
-  elif [[ $(mokutil --sb-state 2>/dev/null) == *"enabled"* ]]; then # if Secure boot is enabled
-    if [[ $(mokutil --test-key "$SIGN_DIR/mok.der" 2>/dev/null) != *"already"* ]]; then # if our keys are not already generated/enrolled by the MOK
-      generate_keys
-    fi
-  else
-    echo "WARNING: Secure Boot is not enabled!"
-  fi
-
   if [[ ! -d "$MOD_SRC_DIR" ]]; then
     mkdir -p "$MOD_SRC_DIR"
     cp -R "$PWD/src/" "$MOD_SRC_DIR/src"
   fi
 
+  set -x
+
   cp dkms.conf "$MOD_SRC_DIR/dkms.conf"
   sed -i "s/PACKAGE_VERSION=.*/PACKAGE_VERSION=\"$VERSION\"/" "$MOD_SRC_DIR/dkms.conf"
   dkms add -m "$MODULE_NAME" -v "$VERSION"
-  dkms build -m "$MODULE_NAME" -v "$VERSION"
-  dkms install -m "$MODULE_NAME" -v "$VERSION"
+  dkms build -m "$MODULE_NAME" -v "$VERSION" -k $KERNEL_VERSION --verbose
+  dkms install -m "$MODULE_NAME" -v "$VERSION" -k $KERNEL_VERSION --verbose
 
   # module auto-loading
   echo "acpi_ec" > /etc/modules-load.d/acpi_ec.conf
