diff --git a/install.sh b/install.sh
index fbb83cc..3518f30 100755
--- a/install.sh
+++ b/install.sh
@@ -38,18 +38,6 @@ MOD_SRC_DIR="/usr/src/$MODULE_NAME-$VERSION"
 
 if ! (dkms status 2>/dev/null | grep -q "$MODULE_NAME/${VERSION}.*installed"); then # if the module is already installed in DKMS
 
-  # For Debian
-  if command -v update-secureboot-policy >/dev/null 2>&1; then
-    update-secureboot-policy --new-key
-    update-secureboot-policy --enroll-key
-  elif [[ $(mokutil --sb-state 2>/dev/null) == *"enabled"* ]]; then # if Secure boot is enabled
-    if [[ $(mokutil --test-key "$SIGN_DIR/mok.der" 2>/dev/null) != *"already"* ]]; then # if our keys are not already generated/enrolled by the MOK
-      generate_keys
-    fi
-  else
-    echo "WARNING: Secure Boot is not enabled!"
-  fi
-
   if [[ ! -d "$MOD_SRC_DIR" ]]; then
     mkdir -p "$MOD_SRC_DIR"
     cp -R "$PWD/src/" "$MOD_SRC_DIR/src"
